Sources/TuistAutomation/Utilities/BuildGraphInspector.swift

```diff
@@ -147,9 +147,16 @@ public final class BuildGraphInspector: BuildGraphInspecting {
            let testPlan = scheme.testAction?.testPlans?.first(where: { $0.name == testPlanName }),
            let target = testPlan.testTargets.first(where: { isIncluded($0) })?.target
         {
+            // If a test plan is specified we use that
             return graphTraverser.target(path: target.projectPath, name: target.name)
         } else if let testTarget = scheme.testAction?.targets.first {
+            // If no test plan, we check to see if there are targets directly on the testAction
             return graphTraverser.target(path: testTarget.target.projectPath, name: testTarget.target.name)
+        } else if let testPlan = scheme.testAction?.testPlans?.first,
+                  let target = testPlan.testTargets.first(where: { isIncluded($0) })?.target
+        {
+            // If no testplan and no targets directly, then we use the first test plan
+            return graphTraverser.target(path: target.projectPath, name: target.name)
```

Comment:
This is not exactly right. If no test plan is specified and there is a defined test plan that's marked as default, that should take precedence over test targets defined in the test action. The logic should be as this: https://github.com/tuist/tuist/blob/main/Sources/TuistKit/Services/XcodeBuildService.swift#L143-L152

Reply:
That is different to what the existing logic here was, the existing logic made no attempt to use a default testplan - do you want me to change the logic here to match?

Reply:
I have implemented the logic you mentioned across all the changes here, this will also mean you can now run `tuist test` on a scheme that only has test plans, and now tuist will test the default testplan, whereas previously it would have complained there was nothing to test.